internal/e2e/e2e_test.go

```diff
@@ -0,0 +1,187 @@
+// +build e2e
+
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific la
+
+package e2etest
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	authorizedappdb "github.com/google/exposure-notifications-server/internal/authorizedapp/database"
+	authorizedappmodel "github.com/google/exposure-notifications-server/internal/authorizedapp/model"
+	"github.com/google/exposure-notifications-server/internal/database"
+	publishdb "github.com/google/exposure-notifications-server/internal/publish/database"
+	verifyapi "github.com/google/exposure-notifications-server/pkg/api/v1"
+	"github.com/google/exposure-notifications-server/pkg/secrets"
+	"github.com/google/exposure-notifications-server/pkg/util"
+	"github.com/sethvargo/go-envconfig"
+
+	publishmodel "github.com/google/exposure-notifications-server/internal/publish/model"
+)
+
+type testConfig struct {
+	DbName      string `env:"DB_NAME"`
+	ExposureURL string `env:"EXPOSURE_URL"`
+}
+
+func initConfig(tb testing.TB, c *testConfig) {
+	if err := envconfig.ProcessWith(context.Background(), c, envconfig.OsLookuper()); err != nil {
+		tb.Fatalf("Unable to process environment: %v", err)
+	}
+}
+
+// newTestServer sets up client for local testing
+func newE2ETest(tb testing.TB, dbName string) *database.DB {
+	tb.Helper()
+
+	ctx := context.Background()
+
+	db := database.NewTestDatabase(tb)
+	if dbName != "" && !testing.Short() {
+		dbConfig := &database.Config{}
+		sm, err := secrets.SecretManagerFor(ctx, secrets.SecretManagerTypeGoogleSecretManager)
```

Comment:
Why duplicate this instead of using the existing processor?

Reply:
Good point, will combine the envconfig processing

Reply:
Done